go/src/koding/kites/kloud/utils/object/merge.go

```diff
@@ -0,0 +1,26 @@
+package object
+
+import "reflect"
+
+// Merge overwrites fields in v1 with a value of complementary
+// field in v2, if the field in v2 is non-empty.
+func Merge(v1, v2 interface{}) {
+	vv2 := reflect.ValueOf(v2).Elem()
+	if !vv2.IsValid() {
+		return
+	}
+	vv1 := reflect.ValueOf(v1).Elem()
+	for i := 0; i < vv1.NumField(); i++ {
```

Comment:
It does not set fields recursively, as it would make the implementation more complex and it's not needed for now.

Added a TODO.